rfc/2/index.md

```diff
@@ -0,0 +1,459 @@
+
+In summer 2023, version 3 of the Zarr specification has been accepted by the Zarr implementation and steering councils through the [ZEP process](https://zarr.dev/zeps/accepted/ZEP0001.html).
+A major motivation for the new version is the introduction of extension hooks.
+
+One of these extensions is the sharding codec that has also [been accepted](https://zarr.dev/zeps/accepted/ZEP0002.html) by the Zarr councils.
+Sharding provides a mechanism to store multiple chunks within one file/object.
+This can greatly reduce the number of files/objects that are required for large Zarr arrays, while preserving fast access to individual chunks and parallel writing capabilities.
+Sharding solves some pain points that are also greatly felt by users in the OME-Zarr community.
+Adopting Zarr v3 in OME-Zarr is a precondition for using sharding.
+
+Library support for Zarr v3 is already available for several languages:
+
+- [tensorstore (C++/Python)](https://github.com/google/tensorstore)
+- [zarrita (Python)](https://github.com/scalableminds/zarrita)
+- [zarrita.js (JS)](https://github.com/manzt/zarrita.js/)
+- [zarr3-rs (Rust)](https://github.com/clbarnes/zarr3-rs)
+
+Visualization tools with integrated Zarr implementations are also available:
+
+- [neuroglancer](https://github.com/google/neuroglancer)
+- [WEBKNOSSOS](https://github.com/scalableminds/webknossos)
+
+Support for other languages is under active development, including C, Java and Python.
+
+Libraries will likely prioritize support for v3 over previous versions in the near future.
+OME-Zarr should therefore adopt the new version for future-proofing.
+
+### Sharding
+
+One of the features that become available through the adoption of Zarr v3 is sharding.
+Sharding provides a mechanism where multiple chunks can be stored in a single file/object.
+This can greatly reduce the number of files (i.e. inodes) or objects that are required to store large OME-Zarr images.
+Storing many files/objects can be prohibitive on several storage backends.
+Therefore, sharding (or similar solutions) are a requirement to scale OME-Zarr to peta-scale images.
+
+The sharding mechanism of Zarr v3 is specified in the [sharding codec](https://zarr-specs.readthedocs.io/en/latest/v3/codecs/sharding-indexed/v1.0.html).
+
+![Illustration of a sharded array](https://zarr-specs.readthedocs.io/en/latest/_images/sharding.png)
+
+Each shard contains an index that contains references to the inner chunks that are stored within a shard.
+Inner chunks are compressed individually, if such a codec is specified.
+Implementations can read inner chunks individually.
+Depending on the choice of codecs and the underlying storage backends, it may be possible to write inner chunks individually.
+However, in the general case, writing is limited to entire shards.
+
+## Proposal
+
+This RFC proposes to adopt version 3 of the Zarr format for OME-Zarr.
+Version 2 will no longer be supported.
```

Comment:
In what sense? What will happen with data already in zarr v 2 - will it remain at least readable in foreseeable future?

Reply:
That depends on the implementations. If libraries continue to support OME-Zarr ≤ 0.5, Zarr 2 will remain accessible. As I noted in another section of the proposal, it is fairly easy to migrate Zarr v2 to v3 without rewriting the chunk data.

Reply:
Wording remains unclear -- "no longer be supported" by what? by libraries? by standard itself?

if "easy" -- why not to be instructed/recommended to be done on the fly by the libraries?

My motivation: on dandiarchive.org we already have many TBs of OME-Zarr v2.  Even if it is "easy" to convert, once data is published, and DOI is minted, data must not mutate. 
Ideally a standard should remain stable enough to allow access to data for years coming. Otherwise it would be hard(er) to convince people to place data in such a "standard form" which would not be accessible within just a handful of years.

Reply:
> Wording remains unclear -- "no longer be supported" by what? by libraries? by standard itself?

This means that data that uses OME-Zarr version 0.6 (or whatever the version number ends up being for this RFC) or higher will have to use Zarr v3 instead of v2. Data that uses earlier versions of OME-Zarr are not affected by this requirement.

> if "easy" -- why not to be instructed/recommended to be done on the fly by the libraries?

My hope is that implementations still continue to support a wide range of OME-Zarr versions, including the current 0.4. In the end that is up to each implementation to decide. I think [this table](https://ome.github.io/ome-ngff-tools/) should get another dimension for the OME-Zarr version to track implementation support. I don't think it is in scope for this RFC to make recommendations about support for earlier versions of the spec, though.

> My motivation: on dandiarchive.org we already have many TBs of OME-Zarr v2. Even if it is "easy" to convert, once data is published, and DOI is minted, data must not mutate.
> Ideally a standard should remain stable enough to allow access to data for years coming. Otherwise it would be hard(er) to convince people to place data in such a "standard form" which would not be accessible within just a handful of years.

Since the data in an archive is immutable, it wouldn't be upgraded to new versions of OME-Zarr. That would mean it is not affected by this RFC. If new data is published under the new version of OME-Zarr, it needs to use Zarr v3 under the hood.

There is a case to be made to abandon version numbers and require backwards compatibility on a spec level. Personally, I think that is too early for OME-Zarr. That is something a later RFC could pick up.